### PR TITLE
Test for OAuth2 API, Credentials Membrane Code Flow, Fix for OAuth2 c…

### PR DESCRIPTION
**Pull request by predic8**
…onsent

### CODE REVIEW

File: core/src/main/java/com/predic8/membrane/core/interceptor/authentication/session/LoginDialog.java [hunk 1]
```diff
@@ -13,45 +13,43 @@
...
+import com.floreysoft.jmte.token.*;
+import com.google.common.collect.*;
+import com.predic8.membrane.core.*;
+import com.predic8.membrane.core.exchange.*;
+import com.predic8.membrane.core.http.*;
+import com.predic8.membrane.core.interceptor.*;
+import com.predic8.membrane.core.interceptor.authentication.session.SessionManager.*;
+import com.predic8.membrane.core.interceptor.oauth2.*;
+import com.predic8.membrane.core.interceptor.server.*;
+import com.predic8.membrane.core.resolver.*;
```
Comment (rrayst) [inline core/src/main/java/com/predic8/membrane/core/interceptor/authentication/session/LoginDialog.java]: same change as https://github.com/membrane/service-proxy/pull/445/files#diff-2344c47dfafacd920e4cdec39e19551483b716fc6cab9b0a4973b4bb3a19dd95 ?

File: core/src/test/java/com/predic8/membrane/core/util/URLParamUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.io.*;
+import java.net.*;
+
+import static com.predic8.membrane.core.http.Header.*;
+import static com.predic8.membrane.core.http.MimeType.*;
+import static com.predic8.membrane.core.util.URLParamUtil.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class URLParamUtilTest {
```
Comment (rrayst) [inline core/src/test/java/com/predic8/membrane/core/util/URLParamUtilTest.java:29]: please also add this test to the `UnitTests` suite

**Review by rrayst — APPROVED** [review]

File: core/src/test/java/com/predic8/membrane/core/util/URLParamUtilTest.java (hunk 2 above)
Comment (predic8) [reply core/src/test/java/com/predic8/membrane/core/util/URLParamUtilTest.java:29]: I checked out the branch and the Test is included in the UnitTest? I also pushed.

File: core/src/main/java/com/predic8/membrane/core/interceptor/authentication/session/LoginDialog.java (hunk 1 above)
Comment (predic8) [reply core/src/main/java/com/predic8/membrane/core/interceptor/authentication/session/LoginDialog.java]: ?

**Review by rrayst — APPROVED** [review]

**Pull request merged**